### DISCUSSION TITLE
CTX81Z (Editor for Yamaha TX81Z)

### DISCUSSION THREAD

**Discussion by dnaldoog** [RomanKubiak/ctrlr#469] [Releases]
Anyone got a TX81Z?

I finally finished a panel for the Yamaha TX81Z. Originally it was going to be a simple json file save adaption of [Atom's Demo](https://github.com/RomanKubiak/Panels/blob/1aaeb12ad9cd96a16cddb37fd2a2c8ca98f87a01/RomanKubiak/Yamaha%20-%20TX81Z.bpanelz) but it developed and grew from there.

https://sourceforge.net/projects/ctx81z/

![970](https://user-images.githubusercontent.com/76831683/172087590-a15931a0-a5fd-4c2d-9268-de48a1bfa4e7.png)

**Comment by Godlike-Productions** [comment]
Oooooo pretty.  I wonder if this can be adapted for the Kurzweil Forte/K2700/PC4.  They have a replica of Yamaha's FM stuff inside.  They can even load original DX files.

Projects have a habit of growing.

**Reply by damiensellier** [reply]
@dnaldoog is unstoppable :)

**Reply by dnaldoog** [reply]
> Oooooo pretty. I wonder if this can be adapted for the Kurzweil Forte/K2700/PC4. They have a replica of Yamaha's FM stuff inside. They can even load original DX files.
> 
> Projects have a habit of growing.

Thanks @Godlike-Productions - that’s interesting! Right now I am trying to see how to adapt the panel to the Yamaha V50 (which I have) and also the Yamaha DX 21 (which I don’t have). I think there are a few four op fm synths that have very similar MIDI implementation, so maybe that is one of them?

**Reply by mtarenskeen** [reply]
> I think there are a few four op fm synths that have very similar MIDI implementation

Correct. Similar and compatible MIDI SysEx (but not identical): Yamaha **DX100, DX27, DX27S, DX21, TX81Z, DX11 (=V2 in Japan), WT11, TQ5, YS100, YS200, B200, DS55, V50**. All these synths were supported in YSEDITOR my good old vintage Atari ST editor software from the 80's/90's that I still use from time to time using STeem or Hatari emulator.  STill available on my website [http://yseditor.martintarenskeen.nl](http://yseditor.martintarenskeen.nl)

Nice to have more modern software to edit these good old Yamaha synths though :-)

**Reply by subv0x** [reply]
@dnaldoog have you thought of doing this with the multitimbral mode?

**Comment by dnaldoog** [comment]
> @dnaldoog have you thought of doing this with the multitimbral mode?

hi @subv0x - yes the latest version has that!

![image](https://github.com/RomanKubiak/ctrlr/assets/76831683/3f1bf0d7-aba5-4cd1-8d40-03fb2668e287)

**Reply by subv0x** [reply]
looks great !

**Comment by mtarenskeen** [comment]
You might also want to try my new Ctrlr editor for Yamaha 4-op synths. Mine doesn't support performances and as has a totally different visual approach. But it does support also V50 features like 100-patch bulkdumps, extra V50 voice parameters ( +/- level scaling, low range fixed frequency mode, and DSP effect parameters. You can find it at [https://www.martintarenskeen.nl/ctrlr](url)

**Reply by dnaldoog** [reply]
That link doesn’t work Martin.

it’s a shame there is no longer a central place where we can upload panels to like the original Ctrlr.org website. Otherwise we never know of panels like yours having been completed.

**Reply by damiensellier** [reply]
URL was corrupted : [https://www.martintarenskeen.nl/ctrlr/](https://www.martintarenskeen.nl/ctrlr/)

**Comment by mtarenskeen** [comment]
The link works fine for me?

https://www.martintarenskeen.nl/ctrlr

or

https://martintarenskeen.nl/ctrlr

Can you try again?

Op donderdag 21 december 2023 om 18:33:11 -08:00:00 schreef Damien S. 
***@***.***>:

**Reply by mtarenskeen** [reply]
Sometimes it works if you enter the address manually in your browser, instead of clicking on the link here. But for me both work fine?

**Comment by cv909** [comment]
Im struggling to get the MIDI send rate going, 
what to do when the lowest and max value give the same results - and the 81Z throws overload alerts?

**Reply by dnaldoog** [reply]
Hi @cv909 

there is a midi rate slider at the bottom right corner of the panel. You can adjust that to the right to increase send delay - see if that helps you.

The TX81Z is notorious for being difficult with MIDI mostly because it’s so old.

**Reply by cv909** [reply]
I dragged it all the way to the right, but it didnt change much.
Is there anything inside the 81Z that can be updated/upragded to solve the MIDI transfer issues, or should I just accept it before I ruin the unit? (:

**Reply by dnaldoog** [reply]
The next thing you could try is get a program like  [Bome Send SX](https://www.bome.com/products/sendsx) and send your MIDI out to that program using a [virtual midi port](https://www.tobias-erichsen.de/software/loopmidi.html). On the panel set MIDI out to the virtual port. On BomeSX set MIDI in to the virtual port and MIDI out to the TX81Z. In BomeSX check the **thru** option and somewhere in the settings adjust the transfer rate. 
 
This is for Windows but if you are on macOS I guess you can try something similar.

I connect to the TX81z through a MOTU usb midi timepiece. It’s an old unit but it seems to work perfectly.

P.S ~ if you are using those cheap Chinese USB cables, they don’t work with sysex.

**Reply by cv909** [reply]
im running U-phoria i Steinberg UR 's MIDI in/out - all in  a compat rack so the MIDI cables are 1-1,5 m long and look/feel fairly decent.

**Reply by dnaldoog** [reply]
That sounds like pretty decent equipment. Not sure what to suggest - are you on Windows or macOS?

**Reply by cv909** [reply]
win7 32bit

optionally I may/will try it with win7 64bit Thinkpad if there's any difference.

**Reply by dnaldoog** [reply]
Did you manage to try through _loopMIDI_ and _Bome Send SX_?

**Reply by cv909** [reply]
not yet, 
I tried shorter MIDI cables and it works perfect now.

**Reply by dnaldoog** [reply]
That’s great - possibly one of your cables  was bad?

**Reply by cv909** [reply]
TX must be very picky - out of curiosity Ive put that "bad" cable in between the AlphaJuno 2 and the AlphaBase controller (software) and it works perfect in there, but refuses to transmit data into the 81z - I gotta keep them signed.

and yes, I have to accept I'll never fully understand what's on my hardware's mind (:
cant complain though - being a slave to the midlate 80s synthesizers I;m blessed I can edit them on the fly with the PC, in the 80s I'd probably be forced to use sh101, 303 and the Prophet 5 all of the time.

**Reply by subv0x** [reply]
there's a lot to be said for getting an old Atari ST :-) 
kept mine expressly for that. 
eventually got rid of the CRT monitor when Atari > VGA adapters appeared. (must get an optical mouse for it)
different experience running it with a TFT !

had a great time editing CZ/VZ/TX with it - don't think anyone's made a VZ editor yet. at the time, the cz101 seemed ahead of its time, made me think of digital watches, and supercool, doing a random and seeing what comes out. oh why did i sell that ...

**Comment by dnaldoog** [comment]
> TX must be very picky - out of curiosity Ive put that "bad" cable in between the AlphaJuno 2 and the AlphaBase controller (software) and it works perfect in there, but refuses to transmit data into the 81z - I gotta keep them signed.
> 
> and yes, I have to accept I'll never fully understand what's on my hardware's mind (: cant complain though - being a slave to the midlate 80s synthesizers I;m blessed I can edit them on the fly with the PC, in the 80s I'd probably be forced to use sh101, 303 and the Prophet 5 all of the time.


I also wrote a [Juno panel](https://sourceforge.net/projects/juno-alpha/). That's very interesting about the TX81-Z - it truly must be fussy - probably its age - I think it's the synth I go to most. Recently I put a Sony R7 reverb with it - sounds great!

**Comment by cv909** [comment]
been working like a charm for weeks and I suddenly get some "callback error / command line 57 - "constant copy" / method disabled" error as soon as I try to change waveform for an operator - no idea what happened ;( it's been my best "Regis styled" version of the lately bass and I'd love to edit it more/somehow :)

it only happens with all patches saved in the internal bank - the ABCD allow editing wout errors

//sorted (in a way) somehow the "constant copy" buttons were all activated (pink) - witched them off to their grey state and it works again // phew

**Comment by dnaldoog** [comment]
I think I understand where the error happened- I’ll try and reproduce the error and post a new fix. Thank you.

**Reply by cv909** [reply]
my main problem (not related to the command line 57 in any way)   is the json files that are saved on the pc drive - I cannot load them into the 81z.

I can save my patches into the 81z internal bank using  the "store" / "choose location" function using the ctrlr panel's interface, but I cannot load the (hard drive stored)  json files into the instrument - the panel interface loads ok (the waveforms and envelopes load visually according to what I have saved) but the actual 81z still plays the previously loaded timbre - as if I was missing some "send it!" or "confirm" button to actually load the json settings into the machine.

(I think it would be easier for me to explain it in Polish ;) - Im warsaw based too - I'm definitely doin somethin wrong with the json procedure)

**Comment by dnaldoog** [comment]
I'll have to look into that tomorrow. It's been a while since I looked at the panel, but time has a nasty way of showing the error of one's ways, but _I really appreciate the feedback_. Too often, I guess,  people just work around an error, but if they were to report it then the panel developer could look at it and probably fix it!

Dziękuję

**Reply by cv909** [reply]
all json files saved on my hdd are exactly 6KB - maybe they're incomplete in a way? I run the standalone version on 32bit Win7.

**Comment by dnaldoog** [comment]
Hi @cv909 

I was able to duplicate that error on _line 57_ several times and then suddenly it all seems to be working!

Can you send me your exact steps to reproduce the error?

I think the code is okay, but some control is not being referenced properly when you click on the arrow. I have optimising code for that and I might have to remove that to fix it. Just thinking out aloud here.

Also when I tried loading a JSON saved file back to the panel and sending out to the TX81Z the patch was nothing like what it should be. Does that function **even actually work**? 

I might have to re-write it or abandon it and save files as sysex instead. Will look into it. The code is complex and it has been so long since I wrote it that I am having trouble deciphering what I did! lol.

**Reply by cv909** [reply]
Ive never beeen able to save/load a json file yet.
before clinging to the tx81z Ive used juno2 and JX8P with their ctrlr panels for years and the sysex files work like a charm in every instrument I tried it with - so maybe switching to syx would solve the mystery :)
maybe json is too much windows-dependable? (wild guess)

as for the command like 57 - I'll ttry to get the exact path to the error in the evening.

**Comment by Luciano30003** [comment]
Hi!
(I posted this erroneously onto the main issues, sorry about that...)  

Happy user of this editor, so much fun! Thanks for that!
I'm finding a strange behaviour in ctx81z Tx81z controller, in standalone, runing by itself or as a panel inside ctrlr (the version suggested in sourceforge)
When I touch the coarse or the fine frequency setting in operator one, it sends sysex and keeps sending it until I touch another control or hit sync or snapshot. It gets stuck in a loop. 
Checked on macos with Protokol on M2 mac os monterrey and M2 max ventura.. 
Can anyone reproduce the behaviour or maybe is something I'm doing wrong?
Hope this helps!
Cheers!

**Reply by dnaldoog** [reply]
Hi @Luciano30003 

You are right! I just tested it in Standalone Windows and I get the same behaviour. I will fix now!

**Reply by Luciano30003** [reply]
There you go!!
I was trying to figure out the inner workings of the controller myself without luck. Just beginning in Ctrlr world. Please if you can give me a hint to where to look! (no worries if you don't, I don't mean to bother you!) Maybe when I have more time and patience I hopefully figure out how Ctrlr works and contribute with a tx81z chord effect editor.  But for now I'm super happy with the CTX81Z. Thank you for sharing this!
Cheers!

**Reply by dnaldoog** [reply]
Hi @Luciano30003 it's a bit complex because when I first designed the editor I used a common uiSlider etc for each operator, which was a fundamental design mistake because it was therefore very tricky and convoluted to have DAW automation for each operator. So I created a hidden layer called VST that contains a separate control for each of the four operators. For the fine and coarse settings this hidden modulator was retriggering the MIDI send of the visible control thus creating an infinite loop and endless MIDI send! I imagine the TX81-Z wouldn't be happy as its MIDI implementation is very old and dodgy sometimes!

There's a new panel here. Can you try building in MacOS? I will upload builds from a VirtualBox Mojave as well.

https://sourceforge.net/projects/ctx81z/files/src/

**Comment by dnaldoog** [comment]
I was hoping to hear back from you to hear whether those bug fixes were now working. 

What do you mean by this?

> Maybe when I have more time and patience I hopefully figure out how Ctrlr works and contribute with a tx81z chord effect editor. But for now I'm super happy with the CTX81Z

What do you mean by chord effect editor?

**Comment by Luciano30003** [comment]
Hi!

My apologies for the slow response, I had to finish some work that I had pending.
Indeed the bug is fixed! Tried standalonte and vst on mac.
Yes, I was referring to the chord effect editor. The option to program the chords in chord mode/efffect. But honestly, I think I will need a lot of time to pull it off...maybe it's outside the reach of my skills. And as I was reading, maybe it will be more useful to put that time on fixing the ability to read and write jsons that maybe is of more priority...
Anyway, I use the controller pretty much every day I make music, and it's super fun! So kudos and many many thanks you for your efforts!

Cheers!

**Reply by dnaldoog** [reply]
Great to hear that you're using it to make music every day! Let me know if you find anything else that needs fixing！
